modules/Bio/EnsEMBL/RNAProduct.pm

```diff
@@ -0,0 +1,976 @@
+                  Class-specific maps of attributes to synchronise are
+                  provided by
+                  RNAProductTypeMapper::class_attribute_cache_map() .
+  Returntype    : none
+  Exceptions    : throws if the object contains multiple attributes with the
+                  given code and the choice which one to update is
+                  ambiguous.
+  Caller        : RNAProductAdaptor
+  Status        : At Risk (In Development)
+
+=cut
+
+sub synchronise_attributes {
+  my ($self) = @_;
+
+  my $attribute_cache_map = Bio::EnsEMBL::Utils::RNAProductTypeMapper::mapper()
+    ->class_attribute_cache_map(ref($self));
+
+  while (my ($cache_key, $attr_code) = each %{$attribute_cache_map}) {
+    my $existing_attributes = $self->get_all_Attributes($attr_code);
+    my $n_existing_attrs = scalar @{$existing_attributes};
+    if ($n_existing_attrs > 0) {
+      # At the moment we do not support multiple occurrences of target
+      # attributes at all
+      if ($n_existing_attrs > 1) {
+        throw("Object has multiple '$attr_code' attributes and we do not know"
+              . " which one to update");
+      }
+      else {
+        $existing_attributes->[0]->value($self->{$cache_key});
+      }
+    }
+    else {
+      # No corresponding attribute exists, most likely because we are
+      # dealing with a newly created object which has never been pushed
+      # to the database.
+      $self->add_Attributes(Bio::EnsEMBL::Attribute->new(
+        -CODE  => $attr_code,
+        -VALUE => $self->{$cache_key},
+      ));
+    }
+  }
+
+  return;
+}
+
+
+=head2 transcript
+
+  Arg [1]       : Transcript object (optional)
+  Description   : Sets or retrieves the transcript object associated
+                  with this rnaproduct object.
+  Exceptions    : Throws if there is no adaptor or no dbID defined for
+                  the rnaproduct object.
+  Returntype    : Bio::EnsEMBL::Transcript
+
+=cut
+
+sub transcript {
+  my ($self, $transcript) = @_;
+
+  if (defined($transcript)) {
+
+    # Normal setter
+    assert_ref($transcript, 'Bio::EnsEMBL::Transcript');
+    $self->{'transcript'} = $transcript;
+    weaken($self->{'transcript'});    # Avoid circular references.
+
+  } elsif (@_ > 1) {
+
+    # User has explicitly passed undef. Break connection to transcript.
+    delete( $self->{'transcript'} );
+
+  } elsif (!defined($self->{'transcript'})) {
+    my $adaptor = $self->{'adaptor'};
+    if (!defined($adaptor)) {
+      throw("Adaptor not set for rnaproduct, cannot fetch its transcript");
+    }
+
+    my $dbID = $self->{'dbID'};
+    if (!defined($dbID)) {
+      throw("dbID not set for rnaproduct, cannot fetch its transcript.");
+    }
+
+    $self->{'transcript'} =
+      $adaptor->db()->get_TranscriptAdaptor()
+      ->fetch_by_rnaproduct_id($dbID);
+
+    # Do not weaken the reference if we had to get the transcript from the
+    # database. The user is probably working on rnaproducts directly,
```

Comment:
rnaproduct => RNA product
or should it be $rnaproduct or RNAProduct as used in previous pods?